### PR TITLE
UF-4518: Removed sending status except when in phase investigation an…

### PR DESCRIPTION
**Pull request by MickeWallner**
…d status completion is needed.

### CODE REVIEW

File: src/main/java/se/sundsvall/processengine/parkingpermit/Constants.java [hunk 1]
```diff
@@ -6,6 +6,7 @@ private Constants() {}
 	public static final String PROCESS_ENGINE_USER = "WSO2_Camunda";
 	public static final String PARKING_PERMIT_PROCESS_KEY = "parking-permit";
 	public static final String PHASE_ACTUALIZATION = "Aktualisering";
+	public static final String PHASE_INVESTIGATION = "Utredning";
```
Comment (josc04) [inline src/main/java/se/sundsvall/processengine/parkingpermit/Constants.java:9]: Change to 'Utreda' (I know because I've loaded the diagram into the Camunda Modeller and verified 🙂). Optional is to also add `public static final String PHASE_DESCISION = "Beslut";` to have the complete picture of all phases (for future usage).

File: src/main/java/se/sundsvall/processengine/parkingpermit/integration/casedata/mapper/CaseDataMapper.java [hunk 2]
```diff
@@ -70,17 +70,19 @@ public static StatusDTO toStatus(String statusType, String description) {
...
-	public static StatusDTO toStatusUpdated(ErrandDTO errandDTO){
-		StatusDTO status = new StatusDTO().dateTime(now(ZoneId.systemDefault()));
+	public static Optional<StatusDTO> toStatusUpdated(ErrandDTO errandDTO){
+		if (PHASE_INVESTIGATION.equals(errandDTO.getPhase())) {
+			StatusDTO status = new StatusDTO().dateTime(now(ZoneId.systemDefault()));
 
-		Optional<StatusDTO> isStatusNeedCompletion = errandDTO.getStatuses()
+			Optional<StatusDTO> isStatusNeedCompletion = errandDTO.getStatuses()
 				.stream()
 				.filter(statusDTO -> STATUS_COMPLETION_NEEDED.equals(statusDTO.getStatusType()))
```
Comment (josc04) [inline src/main/java/se/sundsvall/processengine/parkingpermit/integration/casedata/mapper/CaseDataMapper.java:79]: Rename to a constant name that better describes the status, suggestion: `STATUS_AWAITING_COMPLETION`?

File: src/main/java/se/sundsvall/processengine/parkingpermit/integration/casedata/mapper/CaseDataMapper.java [hunk 3]
```diff
@@ -70,17 +70,19 @@ public static StatusDTO toStatus(String statusType, String description) {
...
 				.findAny();
-		if (isStatusNeedCompletion.isPresent()){
-			return status.statusType(STATUS_CASE_UPDATED_COMPLETED);
+			if (isStatusNeedCompletion.isPresent()) {
+				return Optional.of(status.statusType(STATUS_CASE_UPDATED_COMPLETED));
```
Comment (josc04) [inline src/main/java/se/sundsvall/processengine/parkingpermit/integration/casedata/mapper/CaseDataMapper.java:82]: Same here, maybe rename constant to `STATUS_COMPLETION_RECEIVED` to more accurate describe the status

File: src/test/java/se/sundsvall/processengine/parkingpermit/integration/casedata/mapper/CaseDataMapperTest.java [hunk 4]
```diff
@@ -124,15 +123,15 @@ void toStatusUpdatedCaseCompleted(){
...
 	@Test
-	@DisplayName("Should pass if we are not waiting for completion of case and create a status with value STATUS_CASE_PROCESSED")
-	void toStatusUpdatedCaseProcessed(){
+	@DisplayName("Should pass if we are not waiting for completion of case and don't create any status")
+	void toStatusEmpty(){
 		final var statusDTO = toStatusUpdated(TestDataUtility.getTestErrandDTOWithApproval().statuses(emptyList()));
 
-		assertThat(statusDTO.getStatusType()).isEqualTo(STATUS_CASE_PROCESSED);
+		assertThat(statusDTO).isEmpty();
 	}
```
Comment (josc04) [inline src/test/java/se/sundsvall/processengine/parkingpermit/integration/casedata/mapper/CaseDataMapperTest.java:136]: Add test to verify status is empty when phase is not equal to 'Utreda'. Something like:
```
	@Test
	@DisplayName("Should return optional empty when phase other than INVESTIGATION")
	void toStatusUpdatedCaseProcessedForOtherPhaseThanInvestigation() {
		final var statusDTO = toStatusUpdated(TestDataUtility.getTestErrandDTOWithApproval().phase("somePhase"));

		assertThat(statusDTO).isNotPresent();
	}
```

File: src/main/java/se/sundsvall/processengine/parkingpermit/Constants.java (hunk 1 above)
Comment (MickeWallner) [reply src/main/java/se/sundsvall/processengine/parkingpermit/Constants.java:9]: It was "Utreda" at first but I saw "Utredning" in bpmn-file as well. I'll change.

**Review by josc04 — APPROVED** [review]
